### DISCUSSION TITLE
tutorial Darcy simulations not converging from step07d onwards.

### DISCUSSION THREAD

**Discussion by garciapintado** [idaholab/moose#17997] [Q&A Getting Started] [answered]
Dear all,
I am new to MOOSE and cloned it from the repository at around 20-05-2021. My interest is on porous flow simulations.

Going through the Darcy tutorial steps, the simulations have gone very well up to step07_adaptivity, including problems step7a, step7b & step7c. From there onwards things are not so good [using untouched files as they are cloned]:

problem step7d_adapt_blocks.i :: running from command line [on an iMAC] convergence starts nicely, but at time step 15 the convergence starts to struggle. This only get worse, and from step 29 onwards the convergence is terribly slow (~1 minute per time step) and eventually it gets completely stuck.

I've tried to change some Executioner options (e.g. to "solver_type = PJFNK" ) and nothing that I've tried seems able to solve the problem. BTW, in the video tutorial, also Andrew simulation got stuck in this problem, and the he showed results from a stored run. So, I am wondering if there is some current issue going on.

Later on, following problems also get stuck eventually. For example, running [now from peacock with two threads] problem step10_micro.i, the solver goes very well until step 572 (more than half way of the simulation set 1000 timesteps), where it suddenly crashes with the error:

```bash
Nonlinear solve did not converge due to DIVERGED_LINE_SEARCH iterations 10
 Solve Did NOT Converge!
Aborting as solve did not converge

*** ERROR ***
The following error occurred in the object "TimeStepper", of type "ConstantDT".

Solve failed and timestep already at or below dtmin, cannot continue!
```

I am sorry I can find my way around these issues. Please could you help? Has anyone been able to run this simulations lately as they are in the repository?

Best wishes,
Javier

**Comment by garciapintado** [comment, marked as answer]
Dear All,
My fault. Changing the PETSc preconditioning DOES SOLVE the issue. For me, setting the options in the [Executioner] block:
```
[Executioner]
  ...
  petsc_options_iname = '-pc_type -pc_factor_mat_solver'
  petsc_options_value = 'lu mumps'
  ...
[]
```
makes step7d problem in the tutorial converge nicely, and further running it with 4 threads make it run impressively fast in my iMAC.
Best wishes,
Javier

**Reply by aeslaughter** [reply]
Thanks for the update.
